python/cudf_polars/cudf_polars/testing/asserts.py

```diff
@@ -244,3 +249,121 @@ def assert_collect_raises(
     else:
         if cudf_except != ():
             raise AssertionError(f"GPU execution DID NOT RAISE {cudf_except}")
+
+
+def _resolve_sink_format(path: Path) -> str:
+    """Returns valid sink format for assert utilities."""
+    suffix = path.suffix.lower()
+    supported_ext = {
+        ".csv": "csv",
+        ".pq": "parquet",
+        ".parquet": "parquet",
+        ".json": "ndjson",
+        ".ndjson": "ndjson",
+    }
+    if suffix not in supported_ext:
+        raise ValueError(f"Unsupported file format: {suffix}")
+    return supported_ext[suffix]
+
+
+def assert_sink_result_equal(
+    lazydf: pl.LazyFrame,
+    path: str | Path,
+    *,
+    engine: str | GPUEngine = "gpu",
+    read_kwargs: dict | None = None,
+    write_kwargs: dict | None = None,
+) -> None:
+    """
+    Assert that writing a LazyFrame via sink produces the same output.
+
+    Parameters
+    ----------
+    lazydf
+        The LazyFrame to sink.
+    path
+        The file path to use. Suffix must be one of:
+        '.csv', '.parquet', '.pq', '.json', '.ndjson'.
+    engine
+        The GPU engine to use for the sink operation.
+    read_kwargs
+        Optional keyword arguments to pass to the corresponding `pl.read_*` function.
+    write_kwargs
+        Optional keyword arguments to pass to the corresponding `sink_*` function.
+
+    Raises
+    ------
+    AssertionError
+        If the outputs from CPU and GPU sink differ.
+    ValueError
+        If the file extension is not one of the supported formats.
+    """
+    path = Path(path)
+    read_kwargs = read_kwargs or {}
+    write_kwargs = write_kwargs or {}
+
+    fmt = _resolve_sink_format(path)
+
+    cpu_path = path.with_name(f"{path.stem}_cpu{path.suffix}")
+    gpu_path = path.with_name(f"{path.stem}_gpu{path.suffix}")
+
+    sink_fn = getattr(lazydf, f"sink_{fmt}")
+    read_fn = getattr(pl, f"read_{fmt}")
+
+    sink_fn(cpu_path, **write_kwargs)
+    sink_fn(
+        gpu_path,
+        engine=GPUEngine(raise_on_fail=True) if engine == "gpu" else engine,
```

Comment:
Can we use `DEFAULT_EXECUTOR` and `DEFAULT_SCHEDULER` to construct the default engine?

Ref: https://github.com/rapidsai/cudf/blob/191620472f3d3daeadf32003c37ee99eaa4773a9/python/cudf_polars/cudf_polars/testing/asserts.py#L94

Reply:
Yup, good point.